File: python/1636.py
class Solution:
    def frequencySort(self, nums: list[int]) -> list[int]:
        freq = {}
        for num in nums:
            freq[num] = freq.get(num, 0) + 1
        return sorted(nums, key=lambda x: (freq[x], -x))
    
if __name__ == '__main__':
    s = Solution()
    print(s.frequencySort(list(map(int, input().split()))))
        

File: cpp/1636.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<map>

using namespace std;

class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        unordered_map<int, int> freq;
        for(int i : nums)
            freq[i]++;
        sort(nums.begin(), nums.end(), [&](int a, int b){
            return freq[a] == freq[b] ? a > b : freq[a] < freq[b];
        });
        return nums;
    }
};

int main(){
    Solution s;
    int n;
    cin >> n;
    vector<int> nums(n);
    for(int i=0; i<n; i++)
        cin >> nums[i];
    vector<int> ans = s.frequencySort(nums);
    for(int i : ans) 
        cout << i << " ";
    cout << endl;
    return 0;
 
}